src/SFML/Graphics/Text.cpp

```diff
@@ -191,16 +191,38 @@ void Text::setStyle(std::uint32_t style)
 ////////////////////////////////////////////////////////////
 void Text::setFillColor(const Color& color)
 {
-    if (color != m_fillColor)
+    //In case we use the same color on all characters, no need to store
+    //the same color multiple times
+    if (m_fillColor.size() != 1)
+        m_fillColor.resize(1, Color(255, 255, 255));
```

Comment:
```suggestion
        m_fillColor.resize(1, Color::White);
```
Prefer named colors to using the raw RGB codes